java/arcs/sdk/android/storage/service/StorageService.kt

```diff
@@ -165,9 +170,31 @@ open class StorageService : ResurrectorService() {
         }
     }
 
+    override fun onUnbind(intent: Intent): Boolean {
+        if (intent.action == MANAGER_ACTION) {
+            return false
+        }
+
+        val parcelableOptions = requireNotNull(
+            intent.getParcelableExtra<ParcelableStoreOptions?>(EXTRA_OPTIONS)
+        ) { "No StoreOptions found in Intent" }
+
+        val options =
+            parcelableOptions.actual.copy(coroutineContext = coroutineContext)
+
+        val store = stores.remove(options.storageKey)
+
+        cleanupScope.launch {
+            (store?.activate() as? ReferenceModeStore)?.close()
```

Comment:
This looks like it'll work great for ref mode stores on Android, but I think we'll also need a solution (or at least a plan) for how the stores should be closed on non-android Arcs.  Is it good enough to just have StoreManager's closing do the work?

Reply:
Oh, I meant to just add `close` to the `ActiveStore` interface. Then we can implement it appropriate in other store types.

Reply:
I think that for non-Android Arcs, StoreManager (or something similar) could include functionality similar to this (cleanup when no one is using it).

Reply:
Oh, you suggested the `close` change above, I just saw :-)

Reply:
Added some logic in `StoreManager` to trigger the `close` call as well when the `reset` method is used.